src/plugins/ResizeReorder/ReorderHandle.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright Schrodinger, LLC
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReorderHandle
+ * @typechecks
+ */
+
+import React from 'react';
+import cx from 'cx';
+import PropTypes from 'prop-types';
+import DOMMouseMoveTracker from 'DOMMouseMoveTracker';
+import { getState } from 'FixedDataTableStore';
+
+const DRAG_SCROLL_SPEED = 10;
+const DRAG_SCROLL_BUFFER = 100;
+
+class ReorderHandle extends React.Component {
+
+  static propTypes = {
+    height: PropTypes.number,
+    touchEnabled: PropTypes.bool,
+    isRTL: PropTypes.bool,
+    left: PropTypes.number,
+    availableScrollWidth: PropTypes.number,
+    isFixed: PropTypes.bool,
+    scrollToX: PropTypes.func,
+    onColumnReorderEndCallback: PropTypes.func,
+    getCellGroupWidth: PropTypes.func,
+    columnKey: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number
+    ]),
+  };
+
+  /**
+   * Instance of DOMMouseMoveTracker to capture mouse events
+   * @type {DOMMouseMoveTracker}
+   */
+  mouseMoveTracker = null;
+
+  /**
+   * Displacement of reorder cell while reordering
+   * @type {number}
+   */
+  cursorDeltaX = 0;
+
+  /**
+   * Set true while reordering
+   * @type {boolean}
+   */
+  isReordering = false;
+
+  /**
+   * Frame Id of requested animation frame
+   * @type {number | null}
+   */
+  frameId = null;
+
+  /**
+   * Initial position of scroll
+   * @type {number}
+   */
+  scrollStart = 0;
+
+  /**
+   * Left of column
+   * @type {number}
+   */
+  originalLeft = 0;
+
+  render() {
+    const style = {
+      height: this.props.height,
+    };
+
+    return (
+      <div
+        className={cx({
+          'fixedDataTableCellLayout/columnReorderContainer': true,
+          'fixedDataTableCellLayout/columnReorderContainer/active': false,
+        })}
+        onMouseDown={this.onMouseDown}
+        onTouchStart={this.onTouchStart}
+        onTouchEnd={this.onTouchEnd}
+        onTouchMove={this.onTouchMove}
+        style={style}>
+      </div>
+    );
+  }
+
+  onTouchStart = (ev) => {
+    if (!this.props.touchEnabled) {
+      return;
+    }
+    this.onMouseDown(ev);
+  };
+
+  onTouchEnd = (ev) => {
+    if (this.props.touchEnabled)
+      ev.stopPropagation();
+  };
+
+  onTouchMove = (ev) => {
+    if (this.props.touchEnabled)
+      ev.stopPropagation();
+  };
+
+  /**
+   *
+   * @param {MouseEvent} event
+   */
+  onMouseDown = (event) => {
+    this.cursorDeltaX = 0;
+    this.isReordering = true;
+    this.scrollStart = getState().scrollX;
+    this.originalLeft = this.props.left;
+    this.initializeDOMMouseMoveTracker(event);
+    this.updateParentReorderingData({
```

Comment:
Send data to ResizeReorderCell to update the translateX and z-index(through css) style.